lib/routes/initial_walkthrough/initial_walkthrough.dart

```diff
@@ -0,0 +1,258 @@
+import 'dart:io';
+
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:breez/bloc/backup/backup_bloc.dart';
+import 'package:breez/bloc/backup/backup_model.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/logger.dart';
+import 'package:breez/routes/initial_walkthrough/dialogs/beta_warning_dialog.dart';
+import 'package:breez/routes/initial_walkthrough/dialogs/select_backup_provider_dialog.dart';
+import 'package:breez/routes/initial_walkthrough/handlers/backup_restore_handler.dart';
+import 'package:breez/routes/initial_walkthrough/handlers/models/handler.dart';
+import 'package:breez/routes/initial_walkthrough/handlers/models/handler_context_provider.dart';
+import 'package:breez/routes/initial_walkthrough/handlers/restore_request_handler.dart';
+import 'package:breez/routes/initial_walkthrough/loaders/loader_indicator.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/theme_data.dart';
+import 'package:breez/widgets/error_dialog.dart';
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+
+class InitialWalkthroughPage extends StatefulWidget {
+  final Sink<bool> reloadDatabaseSink;
+
+  const InitialWalkthroughPage({this.reloadDatabaseSink});
+
+  @override
+  State<InitialWalkthroughPage> createState() => _InitialWalkthroughPageState();
+}
+
+class _InitialWalkthroughPageState extends State<InitialWalkthroughPage>
+    with HandlerContextProvider, SingleTickerProviderStateMixin {
+  final GlobalKey scaffoldKey = GlobalKey<ScaffoldState>();
+  AnimationController controller;
+  Animation<int> animation;
+
+  final handlers = <Handler>[];
+
+  bool _registered = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    /// Breez Logo Animation
+    controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 2720),
+    )..forward(from: 0.0);
+    animation = IntTween(begin: 0, end: 67).animate(controller)
+      ..addListener(() {
+        setState(() {
+          // The state that has changed here is the animation object’s value
+        });
+      });
+    controller.forward();
+
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      /// Backup Request & Restore Handlers
+      handlers.addAll([
+        RestoreRequestHandler(),
```

Comment:
I don't understand why we need these handlers. Restore is a a process of loading a page with snapshots, picking one and calling the restore endpoint. Can't this be done using standard widgets and pages?
If the concern is about having multiple pages/dialogs (provider, snapshots) we can use a dedicated navigator so we can pop everything all together but currently (unless I am missing something) it looks too complicated to what we are trying to achieve.

Reply:
Main reason was to avoid UI locks and encapsulate the restore request & backup restore flows.

We can but wouldn't it require an overhaul of how these functions communicate both on BackupBloc & Breez library?